x-pack/plugins/viz_editor/common/lib/index_patterns/index_patterns.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import chrome from 'ui/chrome';
+import { IndexPattern } from '../../lib';
+
+export const getIndexPatterns = () => {
+  const savedObjectsClient = chrome.getSavedObjectsClient();
+  return savedObjectsClient
+    .find({
+      type: 'index-pattern',
+      perPage: 1000,
+    })
+    .then(resp => {
+      return resp.savedObjects.map(({ id, attributes }) => {
+        return Object.assign(attributes, {
+          id,
+          title: attributes.title,
+          fields: JSON.parse(attributes.fields as string),
+        }) as IndexPattern;
+      });
+    })
+    .catch(err => {
+      // Log fetch error
```

Comment:
Might want to add a TODO here so it doesn't get lost.

Reply:
Done